player.c

```diff
@@ -391,15 +391,16 @@ static void update_rg_scale(void)
 	db = replaygain_preamp + gain;
 
 	scale = pow(10.0, db / 20.0);
+	check_scale = pow(10.0, (db + player_info_priv.ti->output_gain) / 20.0);
```

Comment:
I felt like this made sense because I use the output gain and R128 as a "combined replaygain" such that players that don't respect R128 tags play at -18 LUFS and those that do play at -23 LUFS + preamp. The decision to calculate a separate scale to be used when checking the limit came from this. I don't know if you suggestion results in the same calculation. I think that the assumed peak at 1.0 idea is good, since that way the combination of output gain + R128 is guaranteed to never exceed the original volume of the file.

Reply:
Besides, what if there exist both an older replaygain peak tag and output gain? Wouldn't it make more sense not to rely on the non-existence of the peak tag when taking the output gain into consideration? I don't know if the standard allows it, but you could probably make some programs tag the file with rg1.0 tags.

Reply:
It's _technically_ possible for an opus file to have a `REPLAYGAIN_TRACK_PEAK` tag but it's not going to happen in practice and not really worth trying to account for in the code. If it did occur, the output gain would probably be 0 anyway, and if both occurred together, would it describe the peak before or after the output gain is applied?

Using check_scale isn't ideal because when it's exceeded we apply `replaygain_scale = limit`, and `limit` won't account for the output_gain. I'm not 100% confident in any of the maths for my suggestion but the results were at least roughly what I was expecting. :shrug:

Reply:
You're right. The limit wouldn't denote the full volume if output_gain is applied. Your idea to calculate the peak based on output_gain is much better. Changed in 967ebe2.